src/RMGGeneratorDecay0.cc

```diff
@@ -43,4 +50,129 @@ void RMGGeneratorDecay0::GeneratePrimaries(G4Event* event) {
   fDecay0G4Generator->GeneratePrimaries(event);
 }
 
+void RMGGeneratorDecay0::BeginOfRunAction(const G4Run*) {
+  // Technically the user can change the seed at any time, so we have to always update.
+  // But only actually update the seed once the user used a RMGGeneratorDecay0 command.
+  if (fUpdateSeeds) {
+    long seed = CLHEP::HepRandom::getTheSeed();
+    // Sanity check
+    if (seed <= 0) {
+      RMGLog::Out(
+          RMGLog::error,
+          "CLHEP seed seems invalid or uninitialized. Setting BxDecay0 seed to 0."
+      );
+      seed = 0;
+    }
+
+    // This should never occur on most systems, as the only way is through the internal CLHEP table
+    // and the values there are all below the max int32.
+    if (seed > std::numeric_limits<int>::max()) {
+      int new_seed = seed - std::numeric_limits<int>::max();
```

Comment:
```suggestion
      long new_seed = seed - std::numeric_limits<int>::max();
```
fixes a clang-tidy warning (we store it in a long variable anyway)